src/System.Private.CoreLib/src/Internal/Reflection/Core/NonPortable/RuntimeConstructedGenericType.cs

```diff
@@ -8,7 +8,10 @@
 using System.Collections.Concurrent;
 
 using Internal.Runtime.Augments;
+
+#if !DISABLE_REFLECTION_TRACE
```

Comment:
Could you please call it `ENABLE_REFLECTION_TRACE`?

`DisableSomething` is naming anti-pattern. The double negation that it typically causes is harder to read and factory for bugs.